### PR TITLE
General: remove backwards compatibility code now that Jetpack requires WP 5.9

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

- General: remove backwards compatibility code for user queries
- Update PHPUnit setup to remove WP < 5.9 oddities (see previous work on this in #21157 and #21175)

**Note**: we will not be able to merge this until #24083 is merged. 

#### Jetpack product discussion

* Primary issue: #24082

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Is CI happy?

### CODE REVIEW

File: docs/examples/bootstrap.php [hunk 1]
```diff
@@ -52,33 +46,6 @@
...
-			"Warning: PHPUnit <9.3 is not compatible with PHP 8.0+, and the hack could not be loaded.\n  Class %s exists: %s\n  File %s exists: %s\n",
-			PHPUnit\Framework\MockObject\InvocationMocker::class,
-			class_exists( PHPUnit\Framework\MockObject\InvocationMocker::class, false ) ? 'yes (bad)' : 'no (good)',
-			"$_tests_dir/includes/phpunit7/MockObject/InvocationMocker.php",
-			file_exists( "$_tests_dir/includes/phpunit7/MockObject/InvocationMocker.php" ) ? 'yes (good)' : 'no (bad)'
-		);
-	}
-}
-
 // Give access to tests_add_filter() function.
```
Comment (anomiex) [inline docs/examples/bootstrap.php:49]: Let's copy in the check for `composer install` having been run that was added to Jetpack in #20392
```suggestion
if ( ! is_readable( $_plugin_root . '/vendor/autoload.php' ) ) {
	echo 'The plugin is not ready for testing.' . PHP_EOL;
	echo PHP_EOL;
	echo 'Composer dependencies must be installed.' . PHP_EOL;
	exit( 1 );
}

// Give access to tests_add_filter() function.
```

**Review by anomiex — COMMENTED** [review]
I started my own PR to do the same thing. 😀 Since you got yours first, I'll just list the extra stuff I found:

* The code trying to load core's speed trap listener needs removal. https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/docs/examples/bootstrap.php#L66-L69
* Docs in monorepo.md need updating too. https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/docs/monorepo.md?plain=1#L237-L238
* Since Gutenberg >8.0 was included in WP 5.6, we can probably take care of this: https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/projects/plugins/jetpack/extensions/blocks/button/edit.js#L46
* This happened back in #19328 and [WP 5.8](https://github.com/WordPress/wordpress-develop/commit/0db14900ff4d5f3996a70ef3b596cc62c95c5192), we can update that and remove the referenced file too. https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/projects/plugins/jetpack/extensions/blocks/story/player/modal/frame.js#L7
* Gutenberg >= 7.2 seems to have happened in WP 5.5. https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/projects/plugins/jetpack/extensions/shared/jetpack-plugin-sidebar.scss#L22
* PHP 5.2 was dropped in WP 5.2, so all this can be reduced to basically line 223 with a variable rename. https://github.com/Automattic/jetpack/blob/cde8ba3af128db2daf061c9a20ac4e0df3e9666c/projects/plugins/jetpack/modules/shortcodes/slideshow.php#L221-L251

If you want I can push these changes to your branch.

**Comment by kraftbj** [comment]
Adding Cherry-Pick label for 10.9.1. Do not merge into the 10.9 branch until after 10.9.0 is stable.

File: projects/plugins/jetpack/modules/shortcodes/slideshow.php [hunk 2]
```diff
@@ -222,8 +222,13 @@ public function slideshow_js( $attr ) {
 		 * Checking for JSON_HEX_AMP and friends here allows us to get rid of
 		 * '&quot;', that can sometimes be included in the JSON input in some languages like French.
 		 */
-		$gallery_attributes = wp_check_invalid_utf8(
-			wp_json_encode( $attr['gallery'], JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT )
+		$gallery_attributes = _wp_specialchars(
```
Comment (anomiex) [inline projects/plugins/jetpack/modules/shortcodes/slideshow.php:225]: I suspect `esc_attr()` might have been enough now. But this works too.

**Review by anomiex — APPROVED** [review]
LGTM as far as it goes. The linting failure should be fixed by #24141, and of course the PHP 8 previous-WP test needs #24083 to make "previous" be 5.9.

**Comment by jeherve** [comment]
r244865-wpcom

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-10.9` in 04b8ad786eb50a64342f972eadafc412df05cefa

**Pull request merged**